ccp/notifications/weeklynotify.py

```diff
@@ -0,0 +1,157 @@
+
+    def body_of_email(self, status, repository):
+        """
+        Generate the body of email for weekly scan notification email
+
+        :param status: Status of build - True=Success False=Failure
+        :type status bool
+        :param repository: Repository name with https:// prefix
+        :type repository str
+        :return: Body of email in text
+        """
+
+        if status:
+            body = self.weekly_body.format(
+                "Scan status:", "Success",
+                "Repository:", repository)
+        else:
+            body = self.weekly_body.format(
+                "Scan status:", "Failure",
+                "Repository:", repository)
+
+        return body + "\n\n" + self.email_footer
+
+    def image_absent_email_body(self):
+        """
+        Generates the body of email for weekly scan notification when
+        image is absent in the registry and scan has failed.
+
+        :return: Body of email in text
+        """
+        body = self.weekly_image_absent_body.format(
+            "Scan status:",
+            "Image is absent in the registry, scan is aborted.")
+        return body + "\n\n" + self.email_footer
+
+    def notify(self,
+               status, namespace, jenkins_url,
+               image_name, build_number, pipeline_name):
+        """
+        Notify the user by sending email about weekly scan info
+
+        :param status: Status of weekly scan performed
+                       ["success", "failed", "image_absent"]
+        :type status str
+        :param namespace: Namespace of the build
+        :type namespace str
+        :param jenkins_url: Jenkins URL to fetch the build info
+        :type jenkins_url str
+        :param image_name: Name of the scanned image
+        :type image_name str
+        :param build_number: Build number of weekly scan at Jenkins
+        :type build_number str / int
+        :param pipeline_name: Pipeline name
+        :type pipeline_name str
+        """
+
+        build_info = self.buildinfo_obj.get_build_info(
+            namespace, jenkins_url,
+            pipeline_name, build_number)
+
+        # possible values are ["success", "failed", "image_absent"]
+        if status == "image_absent":
+            body = self.image_absent_email_body()
+        else:
+            # convert status to boolean
+            status = True if status == "success" else False
+
+            # get the repository name (without tag)
+            repository = "https://" + build_info.get("REGISTRY_URL") + \
+                "/" + image_name.split(":")[0]
+            # body should have repository name (without tag)
+            body = self.body_of_email(status, repository)
+
+        # registry name without ports
+        registry = build_info.get("REGISTRY_URL").strip().split(":")[0]
+
+        # format the subject of email
+        if status and status != "image_absent":
+            subject = self.weekly_success_subj.format(
```

Comment:
Are we running the weekly scan if image is not present? I believe not right?

Reply:
We can't run the scan if image is absent, but we do inform the user that as
```
Scan status:                  Image is absent in the registry, scan is aborted.
```
with subject stating
```
[$registry] FAILED: Weekly scan for nshaikh/build-fail-test:latest
```

code [here](https://github.com/CentOS/container-pipeline-service/pull/632/files#diff-56c0dc4dcf09d1b353acb7a8a6ae508aR58)